server/api/transaction/[id].ts

```diff
@@ -9,6 +9,7 @@ export default defineEventHandler(async (event) => {
       statusMessage: 'Unauthorized',
     });
 
-  const { getForecasts } = useWeatherService(token.googleAccessToken);
-  return await getForecasts();
+  const { getTransaction } = useTransactionService(token.googleAccessToken);
+  const id = event.context.params!.id;
```

Comment:
There is some method like `getParams(event)` to do this safe

Reply:
getRouterParam(event, "id") exists, but you still have to retype that as string for typescript to be happy, I did that